### ISSUE TITLE
SpineOpt Benders algorithm not respecting investments temporal structure

### ISSUE THREAD

**Issue by DillonJ** [spine-tools/SpineOpt.jl#712]
If I create a yearly temporal_block for investments and run a model with rolling operations with a one-year model horizon, I expect a single investment decision covering the whole year, however, the current implementation seems to split this temporal block across the operations windows and I get indepedent investment decisions in each operational window which is not the expected behaviour.

To put this another way - it should be possible to define the temporal structure in such a way that there is a single investment decision common to all operational sub problems - this is the power of Benders

**Comment by manuelma** [comment]
That's right, the way we create the master problem's temporal structure is by rolling the subproblem's temporal structure completely while collecting time slices for each window. Since the subproblem needs to cut temporal blocks to fit within the rolling window, the master problem ends up having cut time-slices too.

But the reason for doing it that way was to make absolutely sure that both master problem and subproblem would be seen the same time slices always, and that there wouldn't be any holes or funny stuff. But I think we handle that situation now anyways by skipping NaNs when creating the benders cuts.

So yes, the more reasonable way to create the master problem's temporal structure now is to just ignore the rolling. Will give it a try now.

**Comment by manuelma** [comment]
I think this is fixed now.
